app/src/main/java/com/example/androidtrainingproject/MainActivity.kt

```diff
@@ -5,33 +5,74 @@ import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
 import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.example.androidtrainingproject.login.LoginScreen
 import com.example.androidtrainingproject.products.ProductDetails
 import com.example.androidtrainingproject.ui.theme.AndroidTrainingProjectTheme
+import dagger.hilt.android.AndroidEntryPoint
 
+@AndroidEntryPoint
 class MainActivity : ComponentActivity() {
     @OptIn(ExperimentalMaterial3Api::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             AndroidTrainingProjectTheme {
                 val navController = rememberNavController()
+
+                var canPop by remember { mutableStateOf(false) }
+                navController.addOnDestinationChangedListener { controller, _, _ ->
+                    canPop = controller.previousBackStackEntry != null
+                }
+
                 Surface(
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
-                    Scaffold {
+                    Scaffold(
+                        topBar = {
+                            if (!canPop) {
```

Comment:
You can either extract the `Text` so it can be reused, or use one `TopAppBar` and conditionally set the `navigationIcon` like `if canPop IconButton(...) else {}`. It's not that it is incorrect, but it can be done in a more efficient way. I know that in the tutorial it was done this way, but it was definitely more due to time constraints and focusing on other, more important topics.